dist/tools/jlink/jlink.sh

```diff
@@ -122,11 +122,10 @@ test_tui() {
     fi
 }
 
-test_opts() {
+test_serial() {
     if [ -n "${JLINK_SERIAL}" ]; then
-        JLINK_OPTS="-SelectEmuBySN '${JLINK_SERIAL}'"
-    else
-        JLINK_OPTS=""
+        JLINK_SERIAL="-SelectEmuBySN '${JLINK_SERIAL}'"
+        JLINK_SERIAL_SERVER="-select usb='${JLINK_SERIAL}'"
```

Comment:
Please swap line 127 and 128 because now you write the updated `JLINK_SERVER` to the `JLINK_SERIAL_SERVER` variable.